### ISSUE TITLE
RAM name isn’t displayed on DELL Latitude D505, D510, D600, D610

### ISSUE THREAD

**Issue by InfoLibre** [memtest86plus/memtest86plus#468]
![IMG_20241125_101858](https://github.com/user-attachments/assets/3d57b937-92dd-4015-8fba-a6ad07213420)

RAM name isn’t displayed on DELL Latitude D505, D510, D600, D610.

[decode-dimms.txt](https://github.com/user-attachments/files/17906823/decode-dimms.txt)

[dmidecode.txt](https://github.com/user-attachments/files/17906824/dmidecode.txt)

[lsmod.txt](https://github.com/user-attachments/files/17906825/lsmod.txt)

[lspci.txt](https://github.com/user-attachments/files/17906826/lspci.txt)

**Comment by debrouxl** [comment]
Hello,
I can't download the `decode-dimms.txt` file ?
But anyway... `lspci.txt` shows that the SMBus controller isn't exposed on the PCI bus, unlike most other standard devices of an ICH6, listed at e.g. https://theretroweb.com/chips/3005 . This is coherent with the `i2c_i801` Linux kernel module not being loaded, according to `lsmod.txt`. Therefore, memtest86+ won't be able to read SPD data under these conditions :)

**Comment by 01e3** [comment]
BTW - https://www.kernel.org/doc/Documentation/i2c/busses/i2c-i801 has a whole section "Hidden ICH SMBus" that discusses the problem. I suppose we could try to unhide the bus which should allows to access the SPD, but I'm not convinced that it would be a good idea - i.e. that would not lead into hangs / crashes. Perhaps conditionally enabled by a flag?

@InfoLibre, if you boot Linux and follow the instruction there, are you able to get it working and read the SPDs?

**Comment by InfoLibre** [comment]
Which commands must I type?

**Comment by InfoLibre** [comment]
$ decode-dimms
No EEPROM found...

$modprobe eeprom
$ decode-dimms
...
Number of SDRAM DIMMs detected and decoded: 0

**Comment by InfoLibre** [comment]
If needed for testing, I can offer and send one of these old laptops.

**Comment by debrouxl** [comment]
From Linux, what's the output of `lspci -nn | grep ISA; setpci -s 00:1f.0 f2.w` on the Latitude D505 pictured above or D600 (seemingly both ICH4*), D510 (seemingly ICH6*), D520 or D610 (seemingly ICH7*). Bonus points if you have some ICH5-based machine.

The `setpci` command was derived from the https://github.com/lm-sensors/lm-sensors/blob/master/prog/hotplug/unhide_ICH_SMBus script mentioned in the documentation pointed by @01e3 . Under that form, it simply reads a value from the PCI bus, so it shall typically be safe; writes could theoretically be another story.

**Comment by InfoLibre** [comment]
DELL Latitude D505
lspci -nn | grep ISA; setpci -s 00:1f.0 f2.w
00:1f.0 ISA bridge [0601]: Intel Corporation 82801DBM (ICH4-M) LPC Interface Bridge [8086:24cc] (rev 01)
ffff


DELL Latitude D520
lspci -nn | grep ISA; setpci -s 00:1f.0 f2.w
00:1f.0 ISA bridge [0601]: Intel Corporation 82801GBM (ICH7-M) LPC Interface Bridge [8086:27b9] (rev 01)
ffff

DELL Latitude D610
lspci -nn | grep ISA; setpci -s 00:1f.0 f2.w
00:1f.0 ISA bridge [0601]: Intel Corporation 82801FBM (ICH6M) LPC Interface Bridge [8086:2641] (rev 03)
ffff


I don't have any other models at the moment but I will post the results as soon as I have any.

**Comment by x86fr** [comment]
Are you able to read the SPD with CPU-Z ?

**Comment by InfoLibre** [comment]
What is SPD?
It takes me a lot of times to install all the laptops, start Linux, do tests and then, pack laptops away. Please ask me ALL tests I must do, not one test after one. After CPU-Z, which tests should I do?

**Comment by InfoLibre** [comment]
Can I see SPD with CPU-X too? (CPU-Z is for Windows)
https://thetumultuousunicornofdarkness.github.io/CPU-X/

**Comment by x86fr** [comment]
Yes, you need Windows to run CPU-Z (even Windows XP is OK). Just launch it and check SPD tab.

If CPU-Z is not able to report SPD, there is nothing we can do (SPD is locked hidden, which is not uncommon on some older Dell/Lenovo). If CPU-Z is able to report SPD, save a report using last tab (or click validate and report the link here if the laptop is connected).

**Comment by InfoLibre** [comment]
**D610**

<img width="1024" height="768" alt="Image" src="https://github.com/user-attachments/assets/1d6b228f-b692-4815-8b46-cc269e859fe2" />

<img width="1024" height="768" alt="Image" src="https://github.com/user-attachments/assets/ef601d72-3410-450f-87b6-db0329ba3e98" />


Report: [610.txt](https://github.com/user-attachments/files/21689120/610.txt)

I used legacy version of CPU-Z, last version displays an error message on Windows XP Pro SP3.

**Comment by InfoLibre** [comment]
**D505**

<img width="1024" height="768" alt="Image" src="https://github.com/user-attachments/assets/52c08750-0b75-45c2-9153-545bdad98e6c" />

<img width="1024" height="768" alt="Image" src="https://github.com/user-attachments/assets/c9c2c956-eef1-40f7-aad3-28c71e78f299" />

Report: [D505.txt](https://github.com/user-attachments/files/21689455/D505.txt)

**Comment by InfoLibre** [comment]
**D520**

<img width="1024" height="768" alt="Image" src="https://github.com/user-attachments/assets/ce685ad8-68a5-4263-9e1a-9162a4c8a54d" />

<img width="1024" height="768" alt="Image" src="https://github.com/user-attachments/assets/ad812d10-50a2-48b7-b767-cd1aa1081adb" />


Report: [D520.txt](https://github.com/user-attachments/files/21689544/D520.txt)

**Comment by debrouxl** [comment]
Thanks. CPU-Z being able to retrieve information shows that the SMBus controller isn't permanently locked out on any of these computers.

**Comment by InfoLibre** [comment]
On D520 only, RAM is displayed:

![Image](https://github.com/user-attachments/assets/fc03bff3-471d-45bc-ae3d-59207a396174)

**Comment by x86fr** [comment]
What's the most convenient way for you to try a fix? binary only? ISO?

**Comment by InfoLibre** [comment]
An ISO is better for me.

**Comment by x86fr** [comment]
Can you please try this on your Dell D505?

[mt.zip](https://github.com/user-attachments/files/21721401/mt.zip)

It's really dirty and must only be launched on ICH3-5 based computers like the D505. 
ICH6+ requires another fix.

**Comment by InfoLibre** [comment]
On D505 now:

![Image](https://github.com/user-attachments/assets/32c74d14-ecb9-48fd-8b9f-e57a6ecb9a4b)

**Comment by debrouxl** [comment]
I have a couple ICH4 platforms, as well as an ICH6 platform. However, despite two of the computers being Dell models (Optiplex GX260, Precision 370), last time I tried, Linux detected the SMBus controller out of the box on all three computers, and I could retrieve SPD data from all memory sticks. At best, I could test that whatever the fixes in memtest86+ are, RAM SPD display [still] works.

**Comment by x86fr** [comment]
Here is a new build with proper implementation of the fix. 

Can you please confirm that fix is still working in D505 (RAM displayed properly)?

Also, I've added a new quirk for the D610 (and only this model). Not sure about this one as I have no way to test it, but it would be great to check.

Here is an ISO built from this branch: https://github.com/memtest86plus/memtest86plus/compare/main...x86fr/smb_unhide

[mt-smbunhide-2025082401.zip](https://github.com/user-attachments/files/21957851/mt-smbunhide-2025082401.zip)

**Comment by InfoLibre** [comment]
D505 with this new version:

![Image](https://github.com/user-attachments/assets/df370a6b-02ce-40ad-83ff-89cc6689ff8b)

RAM name is Unknow because it's no name RAM:

![Image](https://github.com/user-attachments/assets/1c922333-a494-4164-912d-6844d6a8c721)

Do you want I test on D520 and D610?

**Comment by x86fr** [comment]
D520 should be fine as it was already working previously. A test on D610 would be great. About D600, if you still have it, a CPU-Z dump is required if it doesn't work with the current fixes

**Comment by InfoLibre** [comment]
D610 with this last version:

![Image](https://github.com/user-attachments/assets/a83adefc-b163-4909-be0a-5b17f320dd7f)

I do not have anymore D600 but I will test as soon I have one again.

**Comment by InfoLibre** [comment]
I tested https://github.com/user-attachments/files/21957851/mt-smbunhide-2025082401.zip on a Dell Latitude D600, it works:

![Image](https://github.com/user-attachments/assets/bfc88369-e360-434f-a696-a2bc95a5dd7f)

[dmidecode.txt](https://github.com/user-attachments/files/23711995/dmidecode.txt)

[lsmod.txt](https://github.com/user-attachments/files/23711994/lsmod.txt)

[lspci.txt](https://github.com/user-attachments/files/23711996/lspci.txt)

<img width="409" height="409" alt="Image" src="https://github.com/user-attachments/assets/52afe82a-1476-4328-b22a-59ebc7227824" />

[cpu-z.txt](https://github.com/user-attachments/files/23712064/cpu-z.txt)

**Comment by debrouxl** [comment]
Good to know, thanks for the report :)
